smart_contract/executor/src/default.rs

```diff
@@ -192,21 +196,28 @@ pub mod domain {
     ) {
         let domain_id = isi.object_id();
 
-        if is_genesis(executor) {
-            execute!(executor, isi);
-        }
-        match is_domain_owner(domain_id, authority) {
-            Err(err) => deny!(executor, err),
-            Ok(true) => execute!(executor, isi),
-            Ok(false) => {}
-        }
-        let can_unregister_domain_token = tokens::domain::CanUnregisterDomain {
-            domain_id: domain_id.clone(),
-        };
-        if can_unregister_domain_token.is_owned_by(authority) {
+        if is_genesis(executor)
+            || match is_domain_owner(domain_id, authority) {
+                Err(err) => deny!(executor, err),
+                Ok(is_domain_owner) => is_domain_owner,
+            }
+            || {
+                let can_unregister_domain_token = tokens::domain::CanUnregisterDomain {
+                    domain_id: domain_id.clone(),
+                };
+                can_unregister_domain_token.is_owned_by(authority)
+            }
+        {
+            for (owner_id, permission) in accounts_permission_tokens() {
+                if is_token_domain_associated(&permission, domain_id) {
```

Comment:
I feel like it can (or should in future) be implemented using query filter

Reply:
Permission tokens are opaque for core, how it should extract fields?